src/hb-ot-hmtx-table.hh

```diff
@@ -190,33 +190,19 @@ struct hmtxvmtx
 		      unsigned int default_advance_ = 0)
     {
       default_advance = default_advance_ ? default_advance_ : hb_face_get_upem (face);
+      ascender = 0; descender = 0; line_gap = 0;
 
-      bool got_font_extents = false;
-      if (T::os2Tag != HB_TAG_NONE)
-      {
-#define USE_TYPO_METRICS (1u<<7)
-	if (0 != (face->table.OS2->fsSelection & USE_TYPO_METRICS))
-	{
-	  ascender = abs (face->table.OS2->sTypoAscender);
-	  descender = -abs (face->table.OS2->sTypoDescender);
-	  line_gap = face->table.OS2->sTypoLineGap;
-	  got_font_extents = (ascender | descender) != 0;
-	}
-      }
+      /* XXX Shouldn't create a font */
+      hb_font_t *font = hb_font_create (face);
```

Comment:
Right.  Then, add an internal API that takes a face, and doesn't scale.  It should return a float.  Oh, yeah, sine we support variations now, we should store float, not int.

Reply:
Done